kevm-pyk/src/kevm_pyk/utils.py

```diff
@@ -18,6 +20,21 @@
 T2 = TypeVar('T2')
 
 
+def get_foundry_toml_profile(foundry_root: Path) -> Dict[str, Any]:
+    """Read the FOUNDRY_PROFILE profile from foundry.toml."""
+    with open(foundry_root / 'foundry.toml', 'rb') as f:
+        whole_toml = tomlkit.load(f)
+    all_profiles = cast(tomlkit.items.Table, whole_toml['profile'])
+    profile = os.getenv('FOUNDRY_PROFILE', default='default')
+    return all_profiles[profile].unwrap()
+
+
+def get_foundry_out_from_root(foundry_root: Path) -> Path:
+    """Get the Foundry out directory."""
+    foundry_toml = get_foundry_toml_profile(foundry_root)
+    return foundry_root / Path(foundry_toml['out'])
+
+
```

Comment:
These should both be properties of the `Foundry` class instead. The first one can be called:

```
@property
def profile(self) -> Dict[str, Any]:
    ...

@property
def out(self) -> Path:
    out_rel_path = self.profile['out'] if 'out' in self.profile else '.'
    return foundry_out / out_rel_path
```

Reply:
Agreed, that's much cleaner.

However, there's a minor annoyance with approach. In a few places, we construct something like
```
br = BugReport(foundry_out / 'bug_reports)
foundry = Foundry(foundry_root, bug_report=br)
```
and thus need to have `foundry_out` available before constructing the `Foundry` object.

A few possible solutions:
- Keep the `get_foundry_out_from_root` utility function just for this case
- Provide a setter `set_bug_report` for the bug_report rather than passing it to the `Foundry` constructor
- Create a separate class `FoundryProject` (or a better name) just to hold the logic for the project location, `foundry.toml`, and any other config details:
```
class FoundryProject:
    root: Path
    toml: Dict[str, Any]

    def __init__(foundry_root: Path) -> None:
        ...

    @property
    def profile(self) -> Dict[str, Any]:
        ...

    @property
    def out(self) -> Path:
        ...

class Foundry:
    _project: FoundryProject
    _bug_report: Optional[BugReport]

    def __init__(
        self,
        project: FoundryProject,
        bug_report: Optional[BugReport] = None,
    ) -> None:
        self._project = project
        self._bug_report = bug_report
```
then do
```
foundry_proj = FoundryProject(foundry_root)
br = BugReport(foundry_proj.out / 'bug_reports')
foundry = Foundry(foundry_proj, bug_report=br)
```

@ehildenb Any thoughts?

Reply:
Let's just put the bug report in the current directory instead of in the `foundry_out` directory.